tutorials/3d/spring_arm.rst

```diff
@@ -0,0 +1,117 @@
+:article_outdated: False
+
+.. _doc_spring_arm:
+
+Third-person camera with spring arm
+===================================
+
+.. video:: video/spring_arm_camera.webm
+   :alt: Camera attached to a spring arm colliding with walls
+   :autoplay:
+   :loop:
+   :muted:
+
+3D games will often have a third-person camera that follows and
+rotates around something such as a player character or a vehicle.
+
+In Godot, this can be done by setting a :ref:`Camera3D <class_Camera3D>` as a child of a node.
+However, if you try this without any extra steps you'll notice that the camera clips through geometry and hides the scene.
+
+This is where the :ref:`SpringArm3D <class_SpringArm3D>` node comes in.
+
+What is a spring arm?
+---------------------
+
+A spring arm has two main components that affect its behavior.
+
+The "length" of the spring arm is how far from its global position to check for collisions:
+
+.. image:: img/spring_arm_position_length.svg
+
+The "shape" of the spring arm is what it uses to check for collisions. The spring arm will "sweep" this shape from its origin out towards its length.
+
+.. image:: img/spring_arm_shape.svg
+
+The spring arm tries to keep all of its children at the end of its length. When the shape collides with something, the children are instead placed at or near that collision point:
+
+.. image:: img/spring_arm_children.svg
+
+Spring arm with a camera
+------------------------
+
+When a camera is placed as a child of a spring arm, a pyramid representing the camera will be used as the shape.
+
+This pyramid represents the **near plane** of the camera:
+
+.. image::: img/spring_arm_camera_shape.svg
+
+.. note:: If the spring arm is given a specific shape, then that shape will **always** be used.
+
+    The camera's shape is only used if the camera is a **direct child** of the spring arm.
+
+    If no shape is provided and the camera is not a direct child, the spring arm will fall back to using a ray cast which is inaccurate for camera collisions and not recommended.
+
+Every physics process frame, the spring arm will perform a motion cast to check if anything is collided with:
+
+.. image:: img/spring_arm_camera_motion_cast.svg
+
+When the shape hits something, the camera will be placed at the collision point:
+
+.. image:: img/spring_arm_camera_collision.svg
+
+Setting up the spring arm and camera
+------------------------------------
+
+Let's add a spring arm camera setup to the platformer demo.
+
+.. note:: You can download the Platformer3D demo on `GitHub <https://github.com/godotengine/godot-demo-projects/tree/master/3d/platformer>`_ or using the `Asset Library <https://godotengine.org/asset-library/asset/2748>`_.
```

Comment:
```suggestion
.. note:: You can download the Platformer 3D Demo on `GitHub <https://github.com/godotengine/godot-demo-projects/tree/master/3d/platformer>`_ or using the `Asset Library <https://godotengine.org/asset-library/asset/2748>`_.
```
Use the actual name of the demo. I tried searching for "Platformer3D" and nothing came up

Reply:
Good catch! It also needs changed in these two spots (where I copied it from 🫣 )

https://github.com/godotengine/godot-docs/blob/84842ebe00b9b60f540349f62e92c7ad10ac74bf/tutorials/physics/ragdoll_system.rst?plain=1#L11-L13

https://github.com/godotengine/godot-docs/blob/84842ebe00b9b60f540349f62e92c7ad10ac74bf/tutorials/physics/soft_body.rst?plain=1#L35-L37

Reply:
Might have been called something else on the asset library page when these were originally written